discovery_engine_core/bindings/src/types/date_time.rs

```diff
@@ -33,100 +36,106 @@ const MIN_MICRO_SECONDS: i64 = -8_334_632_851_200_000_000;
     unsafe {
-        place.write(date_time);
+        place.write(date_time_utc);
     }
 }
 
 /// Returns the number of micro seconds since midnight on January 1, 1970.
 ///
-/// More specifically it's the number of micro seconds since `1970-01-01T00:00:00Z` assuming
-/// the naive date time to be in UTC.
+/// More specifically it's the number of micro seconds since `1970-01-01T00:00:00Z`.
 ///
 /// # Safety
 ///
-/// The pointer must point to a sound initialized `NaiveDateTime` instance.
+/// The pointer must point to a soundly initialized `DateTimeUtc` instance.
 #[no_mangle]
-pub unsafe extern "C" fn get_naive_date_time_micros_since_epoch(
-    naive_date_time: *const NaiveDateTime,
+pub unsafe extern "C" fn get_date_time_utc_micros_since_epoch(
+    date_time_utc: *const DateTimeUtc,
 ) -> i64 {
-    let naive_date_time = unsafe { &*naive_date_time };
-    let sub_micros = naive_date_time.timestamp_subsec_micros();
-    let seconds = naive_date_time.timestamp();
+    let date_time_utc = unsafe { &*date_time_utc };
+    let sub_micros = date_time_utc.timestamp_subsec_micros();
+    let seconds = date_time_utc.timestamp();
     seconds * MICROS_PER_SECOND + i64::from(sub_micros)
 }
 
-/// Alloc an uninitialized `Box<NaiveDateTime>`, mainly used for testing.
+/// Alloc an uninitialized `Box<DateTimeUtc>`, mainly used for testing.
 #[no_mangle]
-pub extern "C" fn alloc_uninitialized_naive_date_time() -> *mut NaiveDateTime {
+pub extern "C" fn alloc_uninitialized_date_time_utc() -> *mut DateTimeUtc {
     super::boxed::alloc_uninitialized()
 }
 
-/// Drops a `Box<NaiveDateTime>`, mainly used for testing.
+/// Drops a `Box<DateTimeUtc>`, mainly used for testing.
 ///
 /// # Safety
 ///
-/// The pointer must represent an initialized `Box<NaiveDateTime>`.
+/// The pointer must represent an initialized `Box<DateTimeUtc>`.
 #[no_mangle]
-pub unsafe extern "C" fn drop_naive_date_time(naive_date_time: *mut NaiveDateTime) {
-    unsafe { super::boxed::drop(naive_date_time) }
+pub unsafe extern "C" fn drop_date_time_utc(date_time_utc: *mut DateTimeUtc) {
+    unsafe { super::boxed::drop(date_time_utc) }
 }
 
 #[cfg(test)]
 mod tests {
-    use chrono::{NaiveDate, Timelike};
+    use std::mem::MaybeUninit;
+
+    use chrono::Timelike;
 
     use super::*;
 
     #[test]
     fn test_max_date_is_supported() {
-        let place = &mut NaiveDate::from_ymd(1, 1, 1).and_hms(1, 1, 1);
-        unsafe { init_naive_date_time_at(place, MAX_MICRO_SECONDS) };
-        let truncated_max = NaiveDateTime::MAX.with_nanosecond(999_999_000).unwrap();
-        assert_eq!(*place, truncated_max);
-
-        let micros = unsafe { get_naive_date_time_micros_since_epoch(&NaiveDateTime::MAX) };
-        unsafe { init_naive_date_time_at(place, micros) };
-        assert_eq!(*place, truncated_max);
+        let mut place = MaybeUninit::uninit();
+        unsafe { init_date_time_utc_at(place.as_mut_ptr(), MAX_MICRO_SECONDS) };
+        let place = unsafe { place.assume_init() };
```

Comment:
:grey_question: so here (and similarly below) presumably you could have followed the same pattern as before but you wanted to emphasize that `place` should be passed here uninitialized?

Reply:
i would even argue that the tests were written wrongly before. they worked because the overwritten parts in `init_date_time_utc_at()` etc don't have an explicit `Drop` implementation. but as the docs of `init_date_time_utc_at()` correctly state, they should only be called on uninitialized memory (to avoid such pitfalls).

Reply:
cool, thanks for clarifying